cluster-sharding/src/main/scala/org/apache/pekko/cluster/sharding/ClusterSharding.scala

```diff
@@ -425,15 +425,19 @@ class ClusterSharding(system: ExtendedActorSystem) extends Extension {
       allocationStrategy: ShardAllocationStrategy,
       handOffStopMessage: Any): ActorRef = {
 
+    val extractEntityId: ShardRegion.ExtractEntityId = { msg =>
+      messageExtractor.entityId(msg) match {
+        case eid => (eid, messageExtractor.entityMessage(msg))
+      }
+    }
+    val extractShardId: ShardRegion.ExtractShardId = msg => messageExtractor.shardId(msg)
```

Comment:
Should this be privare

Reply:
Thanks, i think this PR wasn't correct. I will find another solution fix this.